src/plots/cartesian/set_convert.js

```diff
@@ -67,24 +82,170 @@ module.exports = function setConvert(ax) {
         else return BADNUM;
     }
 
-    function fromLog(v) {
-        return Math.pow(10, v);
+    /*
+     * wrapped dateTime2ms that:
+     * - accepts ms numbers for backward compatibility
+     * - inserts a dummy arg so calendar is the 3rd arg (see notes below).
```

Comment:
you mean the `setConvert` closure? Because each trace can apply its own calendar to this axis and `setConvert` only happens per axis. But at least the axis internals only need to pass the first arg because it defaults to `ax.calendar`.